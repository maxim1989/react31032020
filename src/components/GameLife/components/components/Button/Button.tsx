import styled from '@emotion/styled';

interface ButtonProps {
    selected?: boolean;
    'data-size'?: number;
    'data-percent'?: number;
    'data-operation'?: string;
}

export const Button = styled.button<ButtonProps>`
  background-color: ${props => props.selected ? 'grey' : '#fff'};
  margin-right: 5px;
  width: 100px;
  &:last-child {
      margin-right: 0;
  }
`;