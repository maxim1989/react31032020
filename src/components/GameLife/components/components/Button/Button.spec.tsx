import React from 'react';
import { shallow } from 'enzyme';
import serializer from 'jest-emotion';

import { Button } from './Button';

describe('Button:', () => {
    it('selected:', () => {
        const wrapper = shallow(<Button selected={true}/>);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('not selected:', () => {
        const wrapper = shallow(<Button selected={false}/>);

        expect(wrapper.html()).toMatchSnapshot();
    });
});