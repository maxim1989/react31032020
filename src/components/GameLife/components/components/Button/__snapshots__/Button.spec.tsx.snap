// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button: not selected: 1`] = `"<button selected=\\"\\" class=\\"css-a4u649-Button e2argv40\\"></button>"`;

exports[`Button: selected: 1`] = `"<button selected=\\"\\" class=\\"css-a4u649-Button e2argv40\\"></button>"`;
